salt/states/file.py

```diff
@@ -2414,6 +2423,8 @@ def managed(
                     - source: https://launchpad.net/tomdroid/beta/0.7.3/+download/tomdroid-src-0.7.3.tar.gz
                     - source_hash: md5=79eef25f9b0b2c642c62b7f737d4f53f
 
+            source_hash is ignored if the file hosted is not on a HTTP or FTP server.
```

Comment:
Source has is ignore on HTTPS servers?

Reply:
`https` is also valid, https://github.com/saltstack/salt/blob/3006.x/salt/modules/file.py#L4469